### PR TITLE
Adds a method to easily write journal entries

### PR DESCRIPTION
**Pull request by idlira**
For an owning type+id and a target class

Fixes: SIRI-298

### CODE REVIEW

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/biz/protocol/DelegateJournalData.java [hunk 1]
```diff
@@ -118,6 +122,21 @@ public void setSilent(boolean silent) {
         this.silent = silent;
     }
 
+    /**
+     * Adds an entry to the journal of the given entity.
+     *
+     * @param targetType the type of the entity under which the entity will be written
+     * @param targetId   the id of the entity under which the entity will be written
```
Comment (cschierle) [inline src/main/java/sirius/biz/protocol/DelegateJournalData.java:129]:
```suggestion
     * @param targetType the type of the entity under which the journal entry will be added
     * @param targetId   the id of the entity under which the journal entry will be added
```
Maybe make the wording here less confusing 😅

**Review by cschierle — APPROVED** [review]

File: src/main/java/sirius/biz/protocol/DelegateJournalData.java [hunk 2]
```diff
@@ -118,6 +122,21 @@ public void setSilent(boolean silent) {
...
 
+    /**
+     * Adds an entry to the journal of the given entity.
+     *
+     * @param targetType the type of the entity under which the journal entry will be created
+     * @param targetId   the id of the entity under which the journal entry will be created
+     * @param aClass     the class of the actual entity which was changed
+     * @param changes    the entry to add to the journal, automatically appended by the class's label
+     */
+    public static void createJournalEntry(String targetType, String targetId, Class<?> aClass, String changes) {
```
Comment (andyHa) [inline src/main/java/sirius/biz/protocol/DelegateJournalData.java:133]: Class<?> aClass
```suggestion
    public static void createJournalEntry(String targetType, String targetId, Class<?> contentIdentifierClass, String contentIdentifierId, String changes) {
```

**Review by andyHa — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/biz/protocol/DelegateJournalData.java (hunk 2 above)
Comment (idlira) [reply src/main/java/sirius/biz/protocol/DelegateJournalData.java:133]: For the direct deletion via (like via Mongo), we have no ID to provide if we don't make a lookup before...

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/biz/protocol/DelegateJournalData.java [hunk 3]
```diff
@@ -118,6 +122,30 @@ public void setSilent(boolean silent) {
...
+    /**
+     * Adds an entry to the journal of the given entity.
+     *
+     * @param targetType             the type of the entity under which the journal entity will be written
+     * @param targetId               the id of the entity under which the journal entity will be written
+     * @param contentIdentifierClass the type of the actual entity which was changed
+     * @param contentIdentifierId    the id of the actual entity which was changed
+     * @param changes                the entry to add to the journal, automatically appended by the class's label
+     */
+    public static void createJournalEntry(String targetType,
```
Comment (andyHa) [inline src/main/java/sirius/biz/protocol/DelegateJournalData.java:134]: use this in  `private void createJournalEntry(Supplier<String> messageSupplier)'  below to removed redundant code?

**Review by andyHa — COMMENTED** [review]
...

**Pull request merged**